taskweaver/module/prompt_util.py

```diff
@@ -0,0 +1,54 @@
+from typing import List, Tuple
+
+
+class PromptUtil:
+    DELIMITER_TEMPORAL: Tuple[str, str] = ("{{DELIMITER_START_TEMPORAL}}", "{{DELIMITER_END_TEMPORAL}}")
+
+    @staticmethod
+    def wrap_text_with_delimiter(text, delimiter: Tuple[str, str]) -> str:
+        """Wrap the provided text with the specified start and end delimiters."""
+        return f"{delimiter[0]}{text}{delimiter[1]}"
+
+    @staticmethod
+    def get_all_delimiters() -> List[Tuple[str, str]]:
+        """Get all the delimiters."""
+        return [getattr(PromptUtil, attr) for attr in dir(PromptUtil) if attr.startswith("DELIMITER_")]
+
+    @staticmethod
+    def remove_parts(text: str, delimiter: Tuple[str, str]) -> str:
+        """Remove the parts of the text that are wrapped by the specified delimiters."""
+        while True:
+            # Find the start of the temporal part
+            start_index = text.find(delimiter[0])
+            # Find the end of the temporal part
+            end_index = text.find(delimiter[1])
+
```

Comment:
Maybe could search from `start_index` in case when there is a dangling closure delimiter before other valid pairs to break later removals

Reply:
done